### PR TITLE
Autelis (Pool/Spa) tests

### PR DESCRIPTION
**Pull request by IgorProtsenkoOd**

### CODE REVIEW

File: src/e2e-tests/page-objects/autelis.page.js [hunk 1]
```diff
@@ -1,11 +1,211 @@
...
+    get offButton() {return $('//button[text()[contains(., "OFF")]]')}
+    get poolButton() {return $('//button[text()[contains(., "POOL")]]')}
+    get spaButton() {return $('//button[text()[contains(., "SPA")]]')}
+    get switchMessageMain() {return $('#autelis-tabs-tabpane-autelis>div>div:first-of-type>div:nth-of-type(2)>div:nth-of-type(2)')}
+    get switchMessageSolar() {return $('#autelis-tabs-tabpane-autelis>div>div:first-of-type>div:nth-of-type(3)>div:nth-of-type(2)')}
 
-    get pageContent() { return $('//div[contains(@class, "ion-page")]') }
+    get solarOnButton() {return $('#autelis-tabs-tabpane-autelis>div>div>div:nth-of-type(3)>div>button:first-of-type')}
+    get solarOffButton() {return $('#autelis-tabs-tabpane-autelis>div>div>div:nth-of-type(3)>div>button:last-of-type')}
+    get cleanerOnButton() {return $('#autelis-tabs-tabpane-autelis>div>div>div:nth-of-type(4)>div>button:first-of-type')}
```
Comment (popescunsergiu) [inline src/e2e-tests/page-objects/autelis.page.js]: A lighter selector here can be: `//*[@id="autelis-tabs-tabpane-autelis"]//div[text()="Cleaner"]/parent::*//button[text()="On"]`
Please don't rely on element positions, they can change even on a framework version upgrade.

This is a general situation

Comment (IgorProtsenkoOd) [reply src/e2e-tests/page-objects/autelis.page.js]: Updated

Comment (popescunsergiu) [reply src/e2e-tests/page-objects/autelis.page.js]: Maybe you forgot to push? the locators are the same as before

File: src/e2e-tests/features/autelis.feature [hunk 2]
```diff
@@ -0,0 +1,96 @@
+@autelis
+Feature: RoboDomo Autelis page
+
```
Comment (popescunsergiu) [inline src/e2e-tests/features/autelis.feature]: Please add a description to this feature page. Usually the scope of the scenarios

File: src/e2e-tests/features/autelis.feature [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        Then Autelis tab shall be selected
+        When User clicks on Autelis tab
+        When User clicks on <offButton> button on Autelis page
+        Then All Autelis SPA-POOL parameters should be turned off
+        When User clicks on <poolButton> button on Autelis page
+        Then Pool parameters are displayed
+        When User clicks on <spaButton> button on Autelis page
+        Then Spa parameters are displayed
+        And Solar parameters are displayed
+        Examples:
```
Comment (popescunsergiu) [inline src/e2e-tests/features/autelis.feature]: if you want this test to be executed only once then `Outline` is not needed. The problem here is that if one fails the rest are not tested

If you want to test these options independently then here is a solution:
```
Scenario Outline: Check Autelis pool/spa/solar options
        Given User loads the RoboDomo web app
        When User clicks Autelis menu button
        Then User is redirected to Autelis page
        Then Autelis tab shall be selected
        When User clicks on Autelis tab
        When User clicks on <option> button on Autelis page
        Then Autelis <parameters> parameters should be turned off
        Examples:
            | option  | parameters |
            | OFF      | SPA-POOL|
            |POOL    | POOL |
            |SPA        | SPA-SOLAR |
```

Also, if you choose this approach then please rework the step definitions also.
Thank you

File: src/e2e-tests/features/autelis.feature [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+            | offButton  | poolButton | spaButton  |
+            | OFF        | POOL       | SPA        |
+
+    @S1 @automated
+    Scenario Outline: Check Solar, Cleaner, Pool Heat, Pool light, Waterfall options 'On' and 'Off'
```
Comment (popescunsergiu) [inline src/e2e-tests/features/autelis.feature]: Same for this Scenario as the above. Please decide on a common or independent approach. If independent then please remove the Outline.

File: src/e2e-tests/features/autelis.feature [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+        Then Autelis <waterfallButton> button is enabled
+        When User clicks on <waterfallButton> button to change state to Off on Autelis page
+        Then Autelis <waterfallButton> button is disabled
+        Examples:
+            | solarBtn | cleanerButton | poolHeatButton | poolLightButton | waterfallButton |
+            | Solar    | Cleaner       | Pool Heat      | Pool Light      | Waterfall       |
+
+
+    @S1 @automated
+    Scenario Outline: Check Spa Heat, Jets, Spa Light, Blower options 'On' and 'Off'
```
Comment (popescunsergiu) [inline src/e2e-tests/features/autelis.feature]: Same for this Scenario as the above. Please decide on a common or independent approach. If independent then please remove the Outline.

File: src/e2e-tests/features/autelis.feature [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+        When User clicks on <blowerButton> button to change state to On on Autelis page
+        Then Autelis <blowerButton> button is enabled
+        When User clicks on <blowerButton> button to change state to Off on Autelis page
+        Then Autelis <blowerButton> button is disabled
+        Examples:
+            | spaHeatButton | jetsButton | spaLightButton | blowerButton |
+            | Spa Heat      | Jets       | Spa Light      | Blower       |
+
+    @S1 @automated
+    Scenario: Check Pool/Spa heat adjustments to lower and higher values
```
Comment (popescunsergiu) [inline src/e2e-tests/features/autelis.feature]: Same for this Scenario as the above. Please decide on a common or independent approach.  Pool and Spa are different areas / devices / sensors, Maybe independently is better

File: src/e2e-tests/features/autelis.feature (hunk 3 above)
Comment (IgorProtsenkoOd) [reply src/e2e-tests/features/autelis.feature]: Updated

File: src/e2e-tests/features/autelis.feature (hunk 6 above)
Comment (IgorProtsenkoOd) [reply src/e2e-tests/features/autelis.feature]: Updated

File: src/e2e-tests/features/autelis.feature (hunk 5 above)
Comment (IgorProtsenkoOd) [reply src/e2e-tests/features/autelis.feature]: Updated

File: src/e2e-tests/features/autelis.feature (hunk 4 above)
Comment (IgorProtsenkoOd) [reply src/e2e-tests/features/autelis.feature]: Updated

**Comment by IgorProtsenkoOd** [comment]
Tested locally and using BrowserStack

**Pull request merged**